api/v1beta1/clusterclass_types.go

```diff
@@ -392,6 +392,7 @@ type PatchSelectorMatch struct {
 
 	// MachineDeploymentClass selects templates referenced in specific MachineDeploymentClasses in
 	// .spec.workers.machineDeployments.
+	// Note: If set to an empty object it will match all MachineDeploymentClass.
```

Comment:
Not sure if we should do this, it's pretty easy for users to match all classes without knowing it this way; have we considered alternative solutions like matching on name globs or labels?

Reply:
/hold

Reply:
To clarify, the glob match should follow https://go.dev/src/path/filepath/match.go

Reply:
I'm also in favor of being explicit in the API and enabling usage of glob inside names allows to do things like `names: [ "*"]` or `names: [ "prod-*"]`

Reply:
Yes, we had exactly this discussion a while back ~ here: https://github.com/kubernetes-sigs/cluster-api/pull/6930#issuecomment-1187187737

Essentially it comes down to:
* intuitive and easily understandable YAML (with using glob) + somewhat consistency with RBAC (but it only supports "*" afaik) vs.
* consistency with "match all" in other Kubernetes selectors, e.g. NetworkPolicies

The PR was implemented as is because we chose consistency. But I'm fine with reconsidering.

Reply:
I'm sorry if we get late to this discussion, but API design and UX are important and require time and broader consensus, and if I'm not wrong, in this case the preferred option is towards intuitive and easily understandable YAML (with using glob)

Reply:
Fine for me

Reply:
> consistency with "match all" in other Kubernetes selectors, e.g. NetworkPolicies

While this is an existing API in the Kubernetes space, it also has a lot of drawbacks when applied to other entities. It's not uncommon for the Cluster API project to different in a few places from what we've seen in other places, in most cases it's when it comes down to explicit behavior or security concerns.

I'm in favor of considering adding support for globs in the current names list.

Reply:
+1

Reply:
I have updated the description https://github.com/kubernetes-sigs/cluster-api/pull/6930#issue-1305732557 with acceptable matches. Can someone confirm if it is true before I start working on the changes?

Reply:
Looks good I'll answer in more detail below

Reply:
> I'm also in favor of being explicit in the API and enabling usage of glob inside names allows to do things like `names: [ "*"]` or `names: [ "prod-*"]`

Do you also expect to match `names:["prod-*", "dev-*"]` or just a single element array?